distribution/lib/Standard/Base/0.0.0-dev/src/Data/Text/Regex/Pattern.enso

```diff
@@ -363,7 +364,8 @@ type Match_Iterator
                 filler_range = Range.new self.cursor match_start
                 filler_span = (Utf_16_Span.Value filler_range self.input)
                 match = Match.Value self.pattern regex_result self.input
-                next_cursor = match.utf_16_end 0
+                ## Handle edge case where match is 0 length
+                next_cursor = (self.cursor + 1).max (match.utf_16_end 0)
```

Comment:
nitpick: I prefer this syntax, IMO its more readable.

```suggestion
                next_cursor = Math.max (self.cursor + 1) (match.utf_16_end 0)
```

Also the Enso comments actually encourage it :)

```
   ? Math.max or Number.max
     While we provide the max method on `Number`, we find it more intuitive to
     write `Math.max a b` rather than `a.max b`. To that end, we recommend using
     the first style.
```